batch2/batch/batch.py

```diff
@@ -280,22 +308,25 @@ def job_record_to_dict(record, running_status=None):
 
     batch_id = record['batch_id']
     job_id = record['job_id']
+    attempt_id = record['n_attempts'] + 1
     id = (batch_id, job_id)
 
     try:
-        body = await job_config(app, record)
+        body = await job_config(app, record, attempt_id)
     except Exception:
         log.exception('while making job config')
         status = {
             'worker': None,
             'batch_id': batch_id,
             'job_id': job_id,
+            'attempt_id': attempt_id,
             'user': record['user'],
             'state': 'error',
             'error': traceback.format_exc(),
             'container_statuses': {k: {} for k in tasks}
         }
-        await mark_job_complete(app, batch_id, job_id, 'Error', status)
+        now = time.time()
+        await mark_job_complete(app, batch_id, job_id, attempt_id, 'Error', status, now, now)
```

Comment:
I'd set timing to None, None to be consistent with cancellation.  It never started running.

Reply:
good catch.